sphinx/highlighting.py

```diff
@@ -50,6 +52,20 @@
 _LATEX_ADD_STYLES = r'''
 \renewcommand\PYGZsq{\textquotesingle}
 '''
+# fix extra space between lines when Pygments highlighting uses \fcolorbox
+# add a {..} to limit \fboxsep scope, and force \fcolorbox use correct value
+# cf pygments #1708 which makes this unneeded for Pygments > 2.7.4
+_LATEX_ADD_STYLES_FIXPYG = r'''
+\makeatletter
+% fix for Pygments <= 2.7.4
+\let\spx@original@fcolorbox\fcolorbox
+\def\spx@fixpyg@fcolorbox{\fboxsep-\fboxrule\spx@original@fcolorbox}
```

Comment:
we don't have to insert here a `\string -\fboxrule` in place of `-\fboxrule` as was done at Pygments 2.8.1 to work-around a minted (latex package using Pygments) compatibility issue